adapters/Mintegral/MintegralAdapter/GADMediationAdapterMintegral.m

```diff
@@ -4,44 +4,98 @@
 // You may obtain a copy of the License at
 //
-//     http://www.apache.org/licenses/LICENSE-2.0
+//     https://www.apache.org/licenses/LICENSE-2.0
 //
 // Unless required by applicable law or agreed to in writing, software
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-//
 
 #import "GADMediationAdapterMintegral.h"
-
 #import "GADMediationAdapterMintegralConstants.h"
+#import "GADMAdapterMintegralExtras.h"
+#import "GADMAdapterMintegralRewardedAdLoader.h"
+#import "GADMAdapterMintegralBannerAdLoader.h"
+#import "GADMAdapterMintegralInterstitialAdLoader.h"
+#import "GADMAdapterMintegralNativeAdLoader.h"
+#import "GADMAdapterMintegralUtils.h"
 
-@implementation GADMediationAdapterMintegral
+#import <MTGSDK/MTGSDK.h>
+#import <MTGSDKBidding/MTGBiddingSDK.h>
 
-- (void)collectSignalsForRequestParameters:(nonnull GADRTBRequestParameters *)params
-                         completionHandler:
-                             (nonnull GADRTBSignalCompletionHandler)completionHandler {
-  // TODO: Start signal generation and call completionHandler when the signal generation finishes or
-  // fails.
-  completionHandler("", nil);
+
+@implementation GADMediationAdapterMintegral {
+    /// Mintegral banner ad.
+    GADMAdapterMintegralBannerAdLoader *_bannerAd;
+
+    /// Mintegral interstitial ad.
+    GADMAdapterMintegralInterstitialAdLoader *_interstitialAd;
+
+    /// Mintegral native ad.
+    GADMAdapterMintegralNativeAdLoader *_nativeAd;
+
+    /// Mintegral rewarded ad.
+    GADMAdapterMintegralRewardedAdLoader * _rewardedAd;
+}
+
+#pragma mark - GADRTBAdapter
++ (nullable Class<GADAdNetworkExtras>)networkExtrasClass {
+    return [GADMAdapterMintegralExtras class];
 }
 
 + (void)setUpWithConfiguration:(nonnull GADMediationServerConfiguration *)configuration
              completionHandler:(nonnull GADMediationAdapterSetUpCompletionBlock)completionHandler {
-  // TODO: Set up Mintegral SDK and perform any necessary prefetching or configuration work.
-  completionHandler(nil);
+    
+    NSMutableSet *appIds = [[NSMutableSet alloc] init];
+    NSMutableSet *appKeys = [[NSMutableSet alloc] init];
+    
+    for (GADMediationCredentials *credential in configuration.credentials) {
+        NSString *appId = credential.settings[GADMAdapterMintegralAppID];
+        GADMAdapterMintegralMutableSetAddObject(appIds, appId);
+        
+        NSString *appKey =credential.settings[GADMAdapterMintegralAppKey];
+        GADMAdapterMintegralMutableSetAddObject(appKeys, appKey);
+    }
+    
+    if (appIds.count < 1 || appKeys.count < 1) {
+        NSError *error = GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Mintegral mediation configurations did not contain a valid app ID or app Key.");
```

Comment:
Nit: Can you update all `app ID` to `App ID` and `app Key` to `App Key`? They should be consistent across all places.

Apply to all the other places.